### ISSUE TITLE
mint DIGG 1:1 for remDIGG

### ISSUE THREAD

**Issue by jorijnsmit** [Badger-Finance/badger-multisig#481]
(original draft for ticket by @mitche50)

NOTE: This doesn't seem as straightforward as initially expected - there is no direct mint function on DIGG so it will need to be upgraded.

If we look at the transfer method, it updates a mapping to move balances
```
    function transfer(address to, uint256 value) public validRecipient(to) returns (bool) {
        uint256 shareValue = value.mul(_sharesPerFragment);
        _shareBalances[msg.sender] = _shareBalances[msg.sender].sub(shareValue);
        _shareBalances[to] = _shareBalances[to].add(shareValue);
        emit Transfer(msg.sender, to, value);
        return true;
    }
```

And if we look at the rebase method, to increase balances the `_totalSupply` is modified then used to update shares per fragment.
```
    function rebase(uint256 epoch, int256 supplyDelta) external onlyMonetaryPolicy onlyAfterRebaseStart returns (uint256) {
        if (supplyDelta == 0) {
            emit LogRebase(epoch, _totalSupply);
            return _totalSupply;
        }

        if (supplyDelta < 0) {
            _totalSupply = _totalSupply.sub(uint256(supplyDelta.abs()));
        } else {
            _totalSupply = _totalSupply.add(uint256(supplyDelta));
        }

        if (_totalSupply > MAX_SUPPLY) {
            _totalSupply = MAX_SUPPLY;
        }

        _sharesPerFragment = TOTAL_SHARES.div(_totalSupply);

        // From this point forward, _sharesPerFragment is taken as the source of truth.
        // We recalculate a new _totalSupply to be in agreement with the _sharesPerFragment
        // conversion rate.
        // This means our applied supplyDelta can deviate from the requested supplyDelta,
        // but this deviation is guaranteed to be < (_totalSupply^2)/(TOTAL_SHARES - _totalSupply).
        //
        // In the case of _totalSupply <= MAX_UINT64 (our current supply cap), this
        // deviation is guaranteed to be < 1, so we can omit this step. If the supply cap is
        // ever increased, it must be re-included.
        // NB: Digg will likely never reach the total supply cap as the total supply of BTC is
        // currently 21 million and MAX_UINT64 is many orders of magnitude greater.
        // _totalSupply = TOTAL_SHARES.div(_sharesPerFragment)

        emit LogRebase(epoch, _totalSupply);
        return _totalSupply;
    }
```

To mint, we will just need to increase the balances of remDIGG holders on `_shareBalances` and update the `_totalSupply` accordingly without modifying the `_sharesPerFragment` variable.

This is exactly what the ERC20 mint function does: https://github.com/Badger-Finance/badger-system/blob/master/deps/openzeppelin-eth/2.0.2/contracts/token/ERC20/ERC20.sol#L175

**Comment by mitche50** [comment]
DIGG minting is not straightforward.  Recommendation is to transfer DIGG from treasury to remDIGG holders and take time to do the mint properly / only allow it to mint to the treasury at all

**Comment by mitche50** [comment]
#520 is a prerequisite.  Once this is done, we will need to call `oneTimeMint()` from the Treasury Ops MSIG in order to mint the DIGG, then distribute to remDIGG holders at 1:1

- [ ] - Upgrade DIGG Proxy
- [ ] - Mint DIGG
- [ ] - Distribute to remDIGG holders

**Comment by jorijnsmit** [comment]
solved atomically in #520
